### ISSUE TITLE
No whitespace allowed in image create -d?

### ISSUE THREAD

**Issue by olgeni** [churchers/vm-bhyve#117]
I tried this:

```
# vm image create -d "Test 2016-09-09" test
/usr/local/sbin/vm: ERROR: 2016-09-09 does not appear to be a valid virtual machine
```

Could it be that an eval somewhere is stripping the quotes?

**Comment by churchers** [comment]
Hmm it looks like the function that parses arguments to the main `vm` command is removing the quotes, so when it gets processed by the `image create` command the space messes things up.

The only solution I can think of is to bypass the `vm` argument processing for the `image` subcommands. Bit ugly but should probably work.

**Comment by churchers** [comment]
Bit late but this should be fixed in 1.3. Might look at applying fix to 1.2 as well although the method for handling command parsing has changed slightly between the versions.

Looks like simply calling `function "$@"` rather than `function $@` is enough to get `sh` to properly quote the individual arguments when passing them around. I originally assumed surrounding `$@` in quotes would just cause it to pass the whole lot as one argument..
